### ISSUE TITLE
Add API Journeys

### ISSUE THREAD

**Issue by andrewvc** [elastic/synthetics#137]
Currently our synthetic journeys are all browser based. In the future it'd be good to add support for browser-less journeys for testing APIs or other things that don't need a browser. 

This would also be useful for supporting odd protocols heartbeat doesn't support since with a full JS environment almost anything can be tested.

This issue is a stub and needs further fleshing out (example syntax etc.)(

**Comment by iorfix** [comment]
Is there any update about this enhancement? Since playwright added test-api-testing, is it already possible to use it in synthetic tests?

**Comment by vigneshshanmugam** [comment]
@iorfix Not yet, we need to expose the request object from Playwright which can be used for API testing.

**Comment by hager-yousri** [comment]
When using playwright document to write api test I always get this error while it is working good with me in the playwright project but it doesn't work in synthetic journeys
TypeError: Cannot read properties of undefined (reading 'post')

**Comment by hager-yousri** [comment]
any update about using playwright api testing in synthetic tests?
if it is still not supported when it is expected to be supported plz?

**Comment by drewpost** [comment]
Hi @hager-yousri - thanks for following this issue. This is a capability that is on our roadmap. We're currently pushing towards the 1.0 release of our new Synthetics app in Elastic Observability. In addition to the full browser synthetic testing, we will be delivering the single ping availability checking that's in Uptime today. After our 1.0 launch, we will look to add in multi-step API journeys. It is a high priority for us.

**Comment by paulmouzas** [comment]
Is there any update on this? I would like to use some lightweight checks that require oauth2 authentication involving an initial step of retrieving an access token from a separate endpoint (OAuth 2 client credentials flow). Heartbeat doesn't support this so a multistep API journey would fulfill this requirement.

I see that there is a request object that does not have the full overhead of the browser object, but it does not appear to be intended to be used in this way. The [docs](https://www.elastic.co/guide/en/observability/current/synthetics-create-test.html#synthetics-request-param) state that 

> The request parameter is not intended to be used for writing pure API tests. Instead, it is a way to support writing plain HTTP requests in service of a browser-based test.

**Comment by paulb-elastic** [comment]
@paulmouzas you are correct in that the `request` object is intended to be used as part of a browser monitor (e.g. getting some content/data from an API endpoint that is then used in a browser based user journey).

I'm afraid we still don't have a timescale for having multi step scriptable lightweight API Journeys at this stage as we focus on the rest of Synthetics (brand new beta Synthetics UI launched in 8.7, multi step browser monitors and single _step_ lightweight checks managed under the [new U](https://www.elastic.co/guide/en/observability/current/synthetics-get-started.html#choose-ui)I and as [Project Monitors](https://www.elastic.co/guide/en/observability/current/synthetics-get-started.html#choose-projects)).

**Comment by iorfix** [comment]
@paulb-elastic  Are there any plans about this feature? It would be very useful for API testing

**Comment by paulb-elastic** [comment]
Hi @iorfix we don't have any more concrete dates on when we'll be getting to this feature, it's on our backlog at the moment, but not prioritised to start working on it.

**Comment by akovov** [comment]
after almost a year from last comment, is there any reprioritization planned ?

**Comment by shahzad31** [comment]
@drewpost @andrewvc @jasonrhodes more requests for prioritising this.
